Tests/StreamChatTests/Audio/AudioSamplesExtractor_Tests.swift

```diff
@@ -0,0 +1,163 @@
+//
+// Copyright © 2023 Stream.io Inc. All rights reserved.
+//
+
+import AVFoundation
+@testable import StreamChat
+import XCTest
+
+final class AudioSamplesExtractor_Tests: XCTestCase {
+    private lazy var subject: AudioSamplesExtractor! = .init()
+    private lazy var sampleBuffer: Data! = .init()
+
+    override func tearDown() {
+        sampleBuffer = nil
+        subject = nil
+        super.tearDown()
+    }
+
+    // MARK: - extractSamples(from:sampleBuffer:samplesPerPixel:)
+
+    func testExtractSamples_withNoSampleBuffer() throws {
+        // Given
+        var sampleBuffer = Data()
+        let readSampleBuffer = makeReadSampleBuffer(
+            data: sampleBuffer,
+            formatDescription: makeFormatDescription()
+        )
+
+        // When
+        let result = subject.extractSamples(
+            from: readSampleBuffer,
+            sampleBuffer: &sampleBuffer,
+            downsamplingRate: 2
+        )
+
+        // Then
+        XCTAssertEqual(result.samplesToProcess, 0)
+        XCTAssertEqual(result.downSampledLength, 0)
+    }
+
+    func testExtractSamples_withValidSampleBuffer_downsamplingIs2_returnsResultWithExpectedProperties() {
+        // Arrange
```

Comment:
Would be nice to have some consistency in the tests structure. On the test above we have Given-When-Then, while here Arrange-Act-Assert. Better to go with G-W-T everywhere.

Reply:
I actually used those comments before write the tests to know what and how to split the code. I prefer the visual separation/grouping where you get all definitions/Given on top then separate with a single line have your execution test then another empty line and then all your assets grouped together. 

For small tests cases like those it's easy to follow and doesn't require the boiler plate of adding comments everywhere. 

Those comments (IMO) are suitable for long testCases with many assets and preconditions (In those cases though it may be better to split the testCase into multiple smaller ones rather than adding comments)